src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> addedNode = new Node<>(value);
+        if (size == 0) {
+            first = addedNode;
+        } else {
+            addedNode.prev = last;
+            last.next = addedNode;
+        }
+        last = addedNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdding(index);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> addedNode = new Node<>(value);
+            addedNode.next = first;
+            first.prev = addedNode;
+            first = addedNode;
+            size++;
+        } else {
+            Node<T> searchedNode = findNodeByIndex(index);
+            Node<T> addedNode = new Node<>(value);
+            addedNode.next = searchedNode;
+            addedNode.prev = searchedNode.prev;
+            searchedNode.prev.next = addedNode;
+            searchedNode.prev = addedNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T node : list) {
+            add(node);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> searchedNode = findNodeByIndex(index);
+        return searchedNode.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = findNodeByIndex(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = findNodeByIndex(index);
+        unlinkNode(node);
+        return node.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> node = first;
+        while (node != null && !((node.value == object) || (node.value != null
+                && node.value.equals(object)))) {
+            node = node.next;
+        }
+        if (node == null) {
+            return false;
+        }
+        unlinkNode(node);
+        return true;
```

Comment:
nested conditions, indented code is harder to read, let's simplify it - 
```suggestion
        Node<T> node = first;
        while (node != null) {
            if (**equals condition**) {
                unlinkNode(node);
                return true;
           }    
           node = node.next;
        }
        return false;
```